pkg/fingerprint/engine.go

```
@@ -0,0 +1,804 @@
+	}
+
+	addr := net.JoinHostPort(t.IP, strconv.Itoa(t.Port))
+
+	var softResult *MatchResult
+
+	probes := e.orderUDPProbes(t.Port)
+	for _, sp := range probes {
+		if ctx.Err() != nil {
+			return nil
+		}
+		conn, err := net.DialTimeout("udp", addr, e.timeout)
```

golangci-lint (2.11.4)

[error] net.DialTimeout must not be called. use (*net.Dialer).DialContext with (*net.Dialer).Timeout

(noctx)